types/vue.d.ts

```diff
@@ -67,8 +68,51 @@ export declare class Vue {
   $nextTick(callback: (this: this) => void): void;
   $nextTick(): Promise<void>;
   $createElement: CreateElement;
+}
+
+export type CombinedVueInstance<Data, Methods, Computed, Props> = Data & Methods & Computed & Props & Vue;
+export type ExtendedVue<Constructor extends VueConstructor, Data, Methods, Computed, Props> =
+  (new (...args: any[]) => CombinedVueInstance<Data, Methods, Computed, Props>) &
```

Comment:
As `ExtendedVue` is a sub-class of the Vue constructor, we can expect the constructor signature would only receive one argument like below?

```ts
new (options: any) => CombinedVueInstance<Data, Methods, Computed, Props>
```

Not sure whether we could annotate the `options` with `ThisTypedComponentOptionsWithArrayProps` etc. since the type declaration might be so complicated 😂